### PR TITLE
Pre experiment tests

### PR DESCRIPTION
**Pull request by sabraaap**
With the following additions / changes:

1. Option to set either constant (randomly sampled) numerical or a constant string-embedding based mission and feedback at inference, or the actual mission; also, if string-based and using random feedback or mission, the string will be replaced with random strings at inference time the same way as at train time
2. Option to pass the loss averaging method as an argument to the train script
3. Require for use_rtg to be set explicitly (rather than assuming this is used by default when not using either feedback or mission, and not used by default when using feedback and/or mission, and then overriding it)
4. For random datasets, create these based on the desired number of seeds and episodes per seed - ablated with the result that the default of 128 seed and 10 episodes per seed works well, but 128 seeds and 100 episodes per seed looks slightly more promising (1280 seeds with 10 episodes each seems to perform the best, but 1280 seeds complicates things and the difference doesn't look statistically significant). Use the product of seeds and episodes per seeds as the dataset length used for sampling batches during training
5. Print the argument values used during each experiment in the slurm log
6. Analysis script that compiles the dataframes for different experiments (and model seeds) into one dataframe and plots the result for a given metric and given arguments (with the option to combine arguments)
7. Changes to argument values and the run name naming logic to allow for easier analysis
8. Changes to argument defaults in line with some preliminary findings (although there may still be changes to this)
9. Additional experiments for hyperparameters including the number of train seeds and episodes per seeds; the episode distribution used for sampling; whether to use random starts for sub-episodes or whether to sample backwards from last step
10. Preliminary experiments to gauge the effect of the type of mission/feedback used at inference and the effect of using random text as feedback or mission

Note: We may have to override the default eps_per_shard for levels with higher max_steps to account for the longer default context length of 64 (maybe try 8, or if that results in CUDA memory errors, 4, and it should work).

### CODE REVIEW

File: src/agent/fdt/base.py [hunk 1]
```diff
@@ -187,23 +188,18 @@ def _forward(self, input: AgentInput):
...
+        criterion = CrossEntropyLoss(reduce=reduce)
         loss = criterion(action_preds, action_targets)
-        return loss
-
-        # criterion = CrossEntropyLoss(reduce=False)
-        # losses = criterion(action_preds, action_targets)
-        # losses = losses.reshape(bacth_size, seq_length)
-        # loss = self._custom_masked_mean_loss(losses, input.attention_mask)
-        # return loss
+        return loss if self.loss_mean_type == "ce_mean" else self._custom_masked_mean_loss(loss.reshape(bacth_size, seq_length), input.attention_mask)
```
Comment (maxtaylordavies) [inline src/agent/fdt/base.py:202]: clean 👍

File: src/trainer/evaluator.py [hunk 2]
```diff
@@ -263,36 +258,23 @@ def _evaluate_ood_generalisation(self, dataset, agent, state: TrainerState):
...
-        #         self.val_seeds.append(
-        #             self._sample_validation_seeds(n=self.num_repeats // 4)
-        #         )
-        # else:
-        self._load_seed_dict(self.collator.dataset, self.collator.dataset.train_config)
-        self.val_seeds.append(self._sample_validation_seeds(self.num_repeats))
+        val_seeds = self._sample_validation_seeds(self.num_repeats)
+        self.val_seeds = val_seeds
+
+    def _set_train_seeds(self):
```
Comment (maxtaylordavies) [inline src/trainer/evaluator.py:264]: cool 👍

**Review by maxtaylordavies — APPROVED** [review]
looks great! nice work 🚀
